### PR TITLE
robertolupi: add my keymaps (q1, q2  knob, preonic) and user files.

### PR DESCRIPTION
**Pull request by robertolupi**
My Keymaps and userspace files, for these keyboards:
- Keychron Q1 ANSI.
- Keychron Q2 ISO with knob (rotary encoder).
- Preonic rev3 drop.

## Description

The userspace code adds support for holding down a modifier for a bit, which I use to keep the Mac task switcher on the screen for a second when using the rotary encoder as a task switcher.

## Types of Changes

- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).

I did not add a readme.md for my keymaps, because I just moved a few keys around.

I read https://docs.qmk.fm/#/documentation_templates and the link to the [Keyboard Layout Editor](https://keyboard-layout-editor.com/) is broken.

- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

I did run make all:robertolupi

### CODE REVIEW

File: keyboards/keychron/q1/rev_0100/keymaps/robertolupi/keymap.c [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    MAC_BASE,
+    MAC_FN,
+    WIN_BASE,
+    WIN_FN,
+};
+
+enum { TD_CAPS };
+
+qk_tap_dance_action_t tap_dance_actions[] = {
+    [TD_CAPS] = ACTION_TAP_DANCE_DOUBLE(CAPS_WORD, KC_CAPS),
```
Comment (drashna) [inline keyboards/keychron/q1/rev_0100/keymaps/robertolupi/keymap.c:30]: I'm not sure that this works, as I believe the "dance double" uses register_code16, which supports only basic (and modded) keycodes.

File: keyboards/keychron/q2/rev_0113/keymaps/robertolupi/config.h [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#pragma once
```
Comment (drashna) [inline keyboards/keychron/q2/rev_0113/keymaps/robertolupi/config.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/keychron/q2/rev_0113/keymaps/robertolupi/keymap.c [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    [_FN2] = LAYOUT_iso_68(
+        KC_GRV,  KC_BRID, KC_BRIU, KC_TASK, KC_FLXP, RGB_VAD, RGB_VAI, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, KC_VOLD,  KC_VOLU,  KC_TRNS,          KC_TRNS, // ...
+        RGB_TOG, RGB_MOD, RGB_VAI, RGB_HUI, RGB_SAI, RGB_SPI, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,                    KC_TRNS,
+        KC_TRNS, RGB_RMOD,RGB_VAD, RGB_HUD, RGB_SAD, RGB_SPD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS, KC_TRNS,
+        QK_BOOT, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS),
+
+};
+
+bool encoder_update_keymap(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/keychron/q2/rev_0113/keymaps/robertolupi/keymap.c:75]: Just a heads up, you may want to switch this to encoder maps.  It supports the layer config like this, bith without extra code.  And it supports ALL keycodes, including custom keycodes.

File: keyboards/preonic/keymaps/robertolupi/keymap.c [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+    return true;
+};
+
+bool muse_mode = false;
+uint8_t last_muse_note = 0;
+uint16_t muse_counter = 0;
+uint8_t muse_offset = 70;
+uint16_t muse_tempo = 50;
+
+bool encoder_update_keymap(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/preonic/keymaps/robertolupi/keymap.c:175]: same here, re: encoder map.

**Comment by robertolupi** [comment]
Thanks for all the tips!

Unfortunately, I found out even more issues with caps word so I need to rethink the whole approach. I will close this PR.

**Pull request closed without merging**